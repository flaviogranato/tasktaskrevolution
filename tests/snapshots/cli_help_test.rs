use std::process::Command;

#[test]
fn test_main_help() {
    let output = Command::new("cargo")
        .args(&["run", "--", "--help"])
        .output()
        .expect("Failed to execute command");

    let stdout = String::from_utf8_lossy(&output.stdout);
    let stderr = String::from_utf8_lossy(&output.stderr);

    // Filter out compilation warnings and focus on the help output
    let help_output = if stdout.contains("Usage: ttr") {
        stdout
    } else {
        stderr
    };

    insta::assert_snapshot!(help_output);
}

#[test]
fn test_create_help() {
    let output = Command::new("cargo")
        .args(&["run", "--", "create", "--help"])
        .output()
        .expect("Failed to execute command");

    let stdout = String::from_utf8_lossy(&output.stdout);
    let stderr = String::from_utf8_lossy(&output.stderr);

    let help_output = if stdout.contains("Create new entities") {
        stdout
    } else {
        stderr
    };

    insta::assert_snapshot!(help_output);
}

#[test]
fn test_create_project_help() {
    let output = Command::new("cargo")
        .args(&["run", "--", "create", "project", "--help"])
        .output()
        .expect("Failed to execute command");

    let stdout = String::from_utf8_lossy(&output.stdout);
    let stderr = String::from_utf8_lossy(&output.stderr);

    let help_output = if stdout.contains("Create a new project") {
        stdout
    } else {
        stderr
    };

    insta::assert_snapshot!(help_output);
}

#[test]
fn test_list_help() {
    let output = Command::new("cargo")
        .args(&["run", "--", "list", "--help"])
        .output()
        .expect("Failed to execute command");

    let stdout = String::from_utf8_lossy(&output.stdout);
    let stderr = String::from_utf8_lossy(&output.stderr);

    let help_output = if stdout.contains("List entities") {
        stdout
    } else {
        stderr
    };

    insta::assert_snapshot!(help_output);
}

#[test]
fn test_validate_help() {
    let output = Command::new("cargo")
        .args(&["run", "--", "validate", "--help"])
        .output()
        .expect("Failed to execute command");

    let stdout = String::from_utf8_lossy(&output.stdout);
    let stderr = String::from_utf8_lossy(&output.stderr);

    let help_output = if stdout.contains("Validate system") {
        stdout
    } else {
        stderr
    };

    insta::assert_snapshot!(help_output);
}
