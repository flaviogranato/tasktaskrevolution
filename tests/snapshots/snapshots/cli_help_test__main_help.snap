---
source: tests/snapshots/cli_help_test.rs
expression: help_output
---
A powerful, flexible, and transparent project management tool that puts you in control. CLI-first design with beautiful HTML outputs.

Usage: ttr [OPTIONS] <COMMAND>

Commands:
  init       Initialize the project management system
  workspace  Initialize workspace with examples (onboarding)
  create     Create new entities
  list       List entities
  update     Update entities
  delete     Delete entities
  link       Link entities
  unlink     Remove links
  report     Generate reports
  validate   Validate system
  build      Build static site
  template   Template management
  task       Task management
  query      Query entities with filtering
  migrate    Migration tools
  serve      Serve HTML files locally
  help       Print this message or the help of the given subcommand(s)

Options:
  -v, --verbose    Enable verbose output (alias for --debug)
      --debug      Enable debug output (alias for --verbose)
  -q, --quiet      Enable quiet output (minimal information)
      --json-logs  Output logs in JSON format
  -h, --help       Print help
  -V, --version    Print version
